kore/src/Kore/Parser/Parser.hs

```diff
@@ -84,41 +82,34 @@ BNF definition:
 
 The @meta-@ version always starts with @#@, while the @object-@ one does not.
 -}
-sortParser :: Parser Sort
-sortParser = do
+objectSortParser :: Parser Sort
+objectSortParser = do
     identifier <- idParser
     c <- ParserUtils.peekChar
     case c of
         Just '{' -> actualSortParser identifier
         _        -> return (SortVariableSort $ SortVariable identifier)
   where
     actualSortParser identifier = do
-        sorts <- inCurlyBracesListParser sortParser
+        sorts <- inCurlyBracesListParser objectSortParser
         return $ SortActualSort SortActual
             { sortActualName = identifier
             , sortActualSorts = sorts
             }
 
-{-|'validateMetaSort' checks that a @meta-sort@ is well-formed.
 
-Relevant BNF definitions:
+metaSortParser :: Parser Sort
+metaSortParser = do
+    sort <- keywordBasedParsers
+        (map
+            (show &&& return . metaSort)
+            metaSortsListWithString
+        )
+    sorts <- inCurlyBracesListParser objectSortParser
```

Comment:
By the way, I think we could replace `objectSortParser` on this line with the `fail` from the next line and get the same error with a more useful location.